src/main/java/com/yelp/nrtsearch/server/luceneserver/warming/WarmingUtils.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.warming;
+
+import com.yelp.nrtsearch.server.grpc.BooleanClause;
+import com.yelp.nrtsearch.server.grpc.BooleanQuery;
+import com.yelp.nrtsearch.server.grpc.ConstantScoreQuery;
+import com.yelp.nrtsearch.server.grpc.DisjunctionMaxQuery;
+import com.yelp.nrtsearch.server.grpc.MultiFunctionScoreQuery;
+import com.yelp.nrtsearch.server.grpc.NestedQuery;
+import com.yelp.nrtsearch.server.grpc.Query;
+import com.yelp.nrtsearch.server.grpc.SearchRequest;
+import com.yelp.nrtsearch.server.grpc.SearchResponse;
+import com.yelp.nrtsearch.server.luceneserver.IndexState;
+import com.yelp.nrtsearch.server.luceneserver.SearchHandler;
+import com.yelp.nrtsearch.server.luceneserver.ShardState;
+import com.yelp.nrtsearch.server.luceneserver.search.SearchContext;
+import com.yelp.nrtsearch.server.luceneserver.search.SearchRequestProcessor;
+import java.io.IOException;
+import org.apache.lucene.facet.taxonomy.SearcherTaxonomyManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WarmingUtils {
+  private static final Logger logger = LoggerFactory.getLogger(WarmingUtils.class);
+
+  public static void handleStrippedWarmingQuery(
```

Comment:
Why do we need the stripped query to go through a different flow? If there is anything else remaining in the query that must be stripped, why can't we just remove those things and go through the same flow?